selene_sdk/train_model.py

```diff
@@ -363,79 +388,43 @@ def _get_batch(self):
-                    "min_loss": min_loss,
-                    "optimizer": self.optimizer.state_dict()
-                }
-                if self.save_new_checkpoints is not None and \
-                        self.save_new_checkpoints >= step:
-                    checkpoint_filename = "checkpoint-{0}".format(
-                        strftime("%m%d%H%M%S"))
-                    self._save_checkpoint(
-                        checkpoint_dict, False, filename=checkpoint_filename)
-                    logger.debug("Saving checkpoint `{0}.pth.tar`".format(
-                        checkpoint_filename))
-                else:
-                    self._save_checkpoint(
-                        checkpoint_dict, False)
-
-            # TODO: Should we have some way to report training stats without running validation?
-            if step and step % self.nth_step_report_stats == 0:
-                logger.info(("[STEP {0}] average number "
-                             "of steps per second: {1:.1f}").format(
-                    step, 1. / np.average(time_per_step)))
-                time_per_step = []
-                valid_scores = self.validate()
-                validation_loss = valid_scores["loss"]
-                self._train_logger.info(train_loss)
-                to_log = [str(validation_loss)]
-                for k in sorted(self._validation_metrics.metrics.keys()):
-                    if k in valid_scores and valid_scores[k]:
-                        to_log.append(str(valid_scores[k]))
-                    else:
-                        to_log.append("NA")
-                self._validation_logger.info("\t".join(to_log))
-                scheduler.step(math.ceil(validation_loss * 1000.0) / 1000.0)
-
-                if validation_loss < min_loss:
-                    min_loss = validation_loss
-                    self._save_checkpoint({
-                        "step": step,
-                        "arch": self.model.__class__.__name__,
-                        "state_dict": self.model.state_dict(),
-                        "min_loss": min_loss,
-                        "optimizer": self.optimizer.state_dict()}, True)
-                    logger.debug("Updating `best_model.pth.tar`")
-                logger.info("training loss: {0}".format(train_loss))
-                logger.info("validation loss: {0}".format(validation_loss))
-
-                # Logging training and validation on same line requires 2 parsers or more complex parser.
-                # Separate logging of train/validate is just a grep for validation/train and then same parser.
-        self.sampler.save_dataset_to_file("train", close_filehandle=True)
+                self._checkpoint()
+            if self.step and self.step % self.nth_step_report_stats == 0:
+                self.validate()
+
+        # We cannot actually support saving training set right?
+        # self.sampler.save_dataset_to_file("train", close_filehandle=True)
```

Comment:
I think at some point we could save training set to file? this was especially useful for sampling on the fly. but the saving made a lot of assumptions (basically it was the BedFileSampler where we saved the coordinates and the indexes of the binary targets) so I wouldn't say it was great :P

Reply:
I will uncomment this line so it will still work for samplers that support `save_dataset_to_file` (it will do nothing for `multi_file_sampler` for now).